envoy/api/v2/core/protocol.proto

```diff
@@ -17,6 +18,14 @@ option (gogoproto.equal_all) = true;
 message TcpProtocolOptions {
 }
 
+message HttpProtocolOptions {
+  // [#not-implemented-hide:]
+  // The idle timeout for upstream connection pool connections. The idle timeout is defined as the
+  // period in which there are no active requests. If not set, there is no idle timeout. When the
+  // idle timeout is reached the connection will be closed.
```

Comment:
So the idle timeout doesn't take into account HTTP/2 pings?  I think that's worth calling out explicitly as it may surprise some users.

Reply:
@alyssawilk I think the wording around "requests" should explain that. But if you want to be more explicit - let me know how you would like to phrase it?

Reply:
"Note that request based timeouts mean that HTTP/2 PINGs will not keep the connection alive."?

I agree that request based timeouts seem fairly clear but I would totally expect folks to ask clarifying questions nonetheless so the more explicit we can be the better :-)